index.js

```diff
@@ -552,90 +551,73 @@ prototype.buildParameter = function(oldParameter) {
 };
 
 /*
- * Convertes Swagger 1.2 authorization definitions to Swagger 2.0 security
- *   definitions
- *
- * @param resourceListing {object} - The Swagger 1.2 Resource Listing document
- * @param convertedSecurityNames {object} - A list of original Swagger 1.2
- * authorization names and the new Swagger 2.0
- *  security names associated with it (This is required because Swagger 2.0 only
- *  supports one oauth2 flow per security definition but in Swagger 1.2 you
- *  could describe two (implicit and authorization_code).  To support this, we
- *  will create a per-flow version of each oauth2 definition, where necessary,
- *  and keep track of the new names so that when we handle security references
- *  we reference things properly.)
+ * Convertes Swagger 1.2 authorization definitions into Swagger 2.0 definitions
+ * Definitions couldn't be converted 1 to 1, 'this.securityNamesMap' should be
+ * used to map between Swagger 1.2 names and one or more Swagger 2.0 names.
  *
+ * @param oldAuthorizations {object} - The Swagger 1.2 Authorizations definitions
  * @returns {object} - Swagger 2.0 security definitions
  */
-prototype.buildSecurityDefinitions = function(resourceListing,
-  convertedSecurityNames) {
-  if (isEmpty(resourceListing.authorizations)) {
-    return undefined;
-  }
-
+prototype.buildSecurityDefinitions = function(oldAuthorizations) {
   var securityDefinitions = {};
 
-  Object.keys(resourceListing.authorizations).forEach(function(name) {
-    var authorization = resourceListing.authorizations[name];
-    var createDefinition = function createDefinition(oName) {
-      var securityDefinition = securityDefinitions[oName || name] = {
-        type: authorization.type
-      };
+  this.securityNamesMap = {};
+  this.forEach(oldAuthorizations, function(oldAuthorization, name) {
+    var scopes = {};
+    this.forEach(oldAuthorization.scopes, function(oldScope) {
+      var name = oldScope.scope;
+      scopes[name] = oldScope.description || ('Undescribed ' + name);
+    });
+
+    var securityDefinition = extend({}, {
+      type: oldAuthorization.type,
+      in: oldAuthorization.passAs,
+      name: oldAuthorization.keyname,
+      scopes: undefinedIfEmpty(scopes)
+    });
 
-      if (authorization.passAs) {
-        securityDefinition.in = authorization.passAs;
+    if (securityDefinition.type === 'basicAuth') {
+      securityDefinition.type = 'basic';
+    }
+
+    if (!isValue(oldAuthorization.grantTypes)) {
+      securityDefinitions[name] = securityDefinition;
+      this.securityNamesMap[name] = [name];
+      return;
+    }
+
+    this.securityNamesMap[name] = [];
+    // For oauth2 types, 1.2 describes multiple "flows" in one authorization object.
+    // But for 2.0 we need to create one security definition per flow.
+    this.forEach(oldAuthorization.grantTypes, function(oldGrantType, gtName) {
+      var grantParameters = {};
+
+      switch (gtName) {
```

Comment:
I think that conversion and validation are totally separate tasks.
For example you can fix Swagger 2.0 after conversion.
But if lib will do throw error that will interrupt conversion you will have no way to convert document.
If user wants to ensure that resulting Swagger document is correct he should use `swagger-tools`.
I think, this is good case for Postel's law:
>Be conservative in what you do, be liberal in what you accept from others

On the other hand there are errors which leading to lost of information like for example [here](https://github.com/apigee-127/swagger-converter/blob/master/index.js#L279)
How should converter react on overriding `basePath`, if it override user will never knew about problem and will be puzzled why some operations doesn't work properly. So instead we should inform user by throwing exception and he could fix problem in original Swagger 1.2 document.

Another example if you expect property to be `array` or `object`  and it actually a `string` you can't perform conversion that mean you should throw, like [here](https://github.com/apigee-127/swagger-converter/blob/master/index.js#L719).

@mohsen1 Are you OK with such error strategy?